app/src/main/java/org/mozilla/focus/fragment/ListPanelDialog.java

```diff
@@ -123,18 +124,27 @@ private void setSelectedItem(int selectedItem) {
     }
 
     private void showDownloads() {
-        setSelectedItem(TYPE_DOWNLOADS);
-        showPanelFragment(DownloadsFragment.newInstance());
+        if(type != TYPE_DOWNLOADS) {
```

Comment:
Let's use getArguments().getInt(TYPE) and don't save a duplicate.

Reply:
If the entry point for tap download tab is MainActivity.showListPanel() -> ListPanelDialog.newInstance() ->ListPanelDialog.showDownloads(), "if(getArguments().getInt(TYPE) != TYPE_DOWNLOADS)" will not be satisfied. So I need to use "type"(default value is 0) as a local reference to cover this case.

Reply:
Ah I see. How about solving it by introducing a member variable `boolean firstLaunch = false` to avoid duplicate entries:

```
    private boolean firstLaunch = true;
    private void showDownloads() {
        if ( firstLaunch || getArguments().getInt(TYPE) != TYPE_DOWNLOADS) {
            setSelectedItem(TYPE_DOWNLOADS);
            showPanelFragment(DownloadsFragment.newInstance());
        }
    }
    private void toggleSelectedItem() {
        firstLaunch = false;
    ......
    }
```

Your code will definitely work too so it's only my personal preference to avoid introducing duplicates (Which makes future developers hard to figure out which to update. )

Let me know if you like yours more and I will merge current implementation.

Reply:
A maybe even better way would be differentiating 1. firstLaunch and 2. Taps by user:
```
    private void showDownloads(boolean firstLaunch) {
        if ( firstLaunch || getArguments().getInt(TYPE) != TYPE_DOWNLOADS) {
            setSelectedItem(TYPE_DOWNLOADS);
            showPanelFragment(DownloadsFragment.newInstance());
        }
    }
```
With this we can even get rid of introducing a new variable but it is slightly confusing the future developers. Personally I like this one the most.